Source/ablastr/fields/IntegratedGreenFunctionSolver.H

```diff
@@ -0,0 +1,225 @@
+/* Copyright 2019-2022 Axel Huebl, Remi Lehe
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#ifndef ABLASTR_IGF_SOLVER_H
+#define ABLASTR_IGF_SOLVER_H
+
+#include <ablastr/constant.H>
+#include <ablastr/utils/Communication.H>
+#include <ablastr/utils/TextMsg.H>
+#include <ablastr/warn_manager/WarnManager.H>
+#include <ablastr/math/fft/AnyFFT.H>
+
+#include <AMReX_Array.H>
+#include <AMReX_Array4.H>
+#include <AMReX_BLassert.H>
+#include <AMReX_Box.H>
+#include <AMReX_BoxArray.H>
+#include <AMReX_Config.H>
+#include <AMReX_DistributionMapping.H>
+#include <AMReX_FArrayBox.H>
+#include <AMReX_FabArray.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_GpuControl.H>
+#include <AMReX_GpuLaunch.H>
+#include <AMReX_GpuQualifiers.H>
+#include <AMReX_IndexType.H>
+#include <AMReX_IntVect.H>
+#include <AMReX_LO_BCTYPES.H>
+#include <AMReX_MFIter.H>
+#include <AMReX_MFInterp_C.H>
+#include <AMReX_MLLinOp.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_Parser.H>
+#include <AMReX_REAL.H>
+#include <AMReX_SPACE.H>
+#include <AMReX_Vector.H>
+
+
+#include <array>
+#include <optional>
+
+
+namespace {
+
+using namespace amrex::literals;
+
+// Implements equation 2 in https://journals.aps.org/prab/pdf/10.1103/PhysRevSTAB.10.129901 with some modification to symmetrize the function
```

Comment:
More robust to rely on DOI :)
```suggestion
// Implements equation 2 in https://doi.org/10.1103/PhysRevSTAB.10.129901 with some modification to symmetrize the function
```

Maybe expand the comment slightly on the modification?

Reply:
It may be too complex to be just a comment in the code. 
I will consider adding this info to the documentation.